delivery_carrier_seur/models/delivery_carrier.py

```diff
@@ -0,0 +1,396 @@
+        selection=[
+            ('002', 'ESTANDAR'),
+            ('004', 'MULTIPACK'),
+            ('006', 'MULTI BOX'),
+            ('018', 'FRIO'),
+            ('052', 'MULTI DOC'),
+            ('054', 'DOCUMENTOS'),
+            ('070', 'INTERNACIONAL ´T'),
+            ('072', 'INTERNACIONAL A'),
+            ('076', 'CLASSIC'),
+            ('118', 'VINO')
+        ],
+        default='002',
+        string='Product code',
+    )
+    seur_send_sms = fields.Boolean(
+        string='Send SMS to customer',
+        help='This feature has delivery cost'
+    )
+    seur_label_format = fields.Selection(
+        selection=[
+            ('pdf', 'PDF'),
+            ('txt', 'TXT'),
+        ],
+        default='pdf',
+        string='Label format',
+    )
+    seur_label_size = fields.Selection(
+        selection=[
+            ('1D', '1D'),
+            ('2D', '2D'),
+            ('2C', '2C'),
+        ],
+        default='2C',
+        string='Label size',
+    )
+    seur_printer = fields.Selection(
+        selection=[
+            ('DATAMAX:E CLASS 4203', 'DATAMAX E CLASS 4203'),
+            ('HP:LASERJET II', 'HP LASERJET II'),
+            ('INTERMEC:C4', 'INTERMEC C4'),
+            ('TEC:B430', 'TEC B430'),
+            ('ZEBRA:LP2844-Z', 'ZEBRA LP2844-Z'),
+            ('ZEBRA:S600', 'ZEBRA S600'),
+            ('ZEBRA:Z4M PLUS', 'ZEBRA Z4M PLUS'),
+        ],
+        string='Printer',
+        default='ZEBRA:LP2844-Z',
+    )
+
+    @api.model
+    def seur_wsdl_get(self, service):
+        if service in ['ImprimirECBWebService', 'IntAppletWebService']:
+            return 'http://cit.seur.com/CIT-war/services/%s?wsdl' % service
+        if service == 'WSConsultaExpediciones':
+            return 'https://ws.seur.com/webseur/services/%s?wsdl' % service
+        raise NotImplementedError
+
+    @api.model
+    def seur_soap_send(self, service, method, data):
+        def trace(title, data):
+            _logger.debug('%s %s: %s' % (
+                method, title, etree.tostring(data['envelope'])))
+
+        def create_rewrite_plugin(data):
+            key = [k for k, v in data.items() if isinstance(v, dict)]
+            if not key:
+                return RewritePlugin('//no-dict', '')
+            key = key[0]
+            if 'total_bultos' not in data[key]:
+                return RewritePlugin('//missing-key', '')
+            xml_root = etree.Element('root')
+            xml_exp = etree.SubElement(xml_root, 'exp')
+            for _ in range(int(data[key].get('total_bultos') or 1)):
+                package = etree.SubElement(xml_exp, 'bulto')
+                for k, v in data[key].items():
+                    etree.SubElement(package, k).text = str(v or '')
+            xml = etree.tostring(xml_root, encoding='utf8', method='xml')
+            data[key] = '-RewritePlugin-'
+            return RewritePlugin('//*[text()="-RewritePlugin-"]', xml)
+
+        history = HistoryPlugin()
+        client = Client(
```

Comment:
¿no sería mejor encapsular una clase SeurRequest() para poder abstraer mejor la capa de conexión?

Reply:
fuera de este modelo no vas a usar los métodos de conexion, entonces que sentido tiene hacerlo más complejo, o dicho de otra forma, que beneficio "real" tendría encapsular estos métodos en una clase.

Reply:
Francamente, creo que es mucho más legible (separación de la parte de SEUR, de la de Odoo), mantenible (cambios futuros de API) y práctico (mucho más fácil hacer mock sobre una clase separada)

Reply:
"Simple is better than complex."
Crear una clase que solo se va a llamar desde un sitio, y con una relación método a método, no ayuda en nada tenerla en otro fichero separado.
Si para que sea más "facil" aplicar futuros cambios de API hay que normalizar los parametros, como puedes ver, serían una gran cantidad de parámetros algo que complicaría en exceso la integración del modelo con la clase encapsulada y viceversa. 

Quizás es que no lo estoy viendo, pero te invito a que hagas un mockup de la clase y los métodos para que pueda ver exactamente como lo aíslas de forma elegante y práctica.

Reply:
La idea es separar en la medida de lo posible la lógica de Odoo, de la de la API correspondiente. Pero bueno, tampoco es algo bloqueante.

Reply:
@chienandalu ya está separada la gestión del calculo de precio en https://github.com/OCA/delivery-carrier/pull/270